plugins/box-whiskers.js

```diff
@@ -60,38 +56,58 @@
                 var screenModel = this.node().screenModel;
                 var model = screenModel.model;
 
-                var cats = this.node().data().reduce(function (memo, row) {
+                var cats = this.node().data().reduce((memo, row) => {
                     var k = row[model.scaleX.dim];
                     memo[k] = memo[k] || [];
                     memo[k].push(row[model.scaleY.dim]);
                     return memo;
                 }, {});
 
-                var MIN = 'minimum';
-                var MAX = 'maximum';
-                var AVG = 'mean';
-                var Q1 = 'Q1';
-                var Q3 = 'Q3';
-
-                var tuples = Object.keys(cats).reduce(function (memo, k) {
-                    var values = cats[k].sort(function (a, b) {
-                        return a - b;
-                    });
+                var tuples = Object.keys(cats).reduce((memo, k) => {
+                    var values = cats[k].sort((a, b) => a - b);
                     var summary = {};
                     summary[model.scaleX.dim] = k;
-                    var sum = values.reduce(function (sum, x) {
-                        return sum + x;
-                    }, 0);
-                    var average = sum / values.length;
-                    var p25 = percentile(values, 0.25);
-                    var p75 = percentile(values, 0.75);
-                    var iqr = p75 - p25;
-
-                    summary[MIN] = (p25 - (iqr * 1.5)); // Math.min.apply(null, values));
-                    summary[MAX] = (p75 + (iqr * 1.5)); // Math.max.apply(null, values));
-                    summary[AVG] = average;
-                    summary[Q1] = p25;
-                    summary[Q3] = p75;
+
+                    var n = values.length;
+                    var q1Pos = n * 0.25;
+                    q1Pos = Math.floor(q1Pos);
+                    var q1 = (q1Pos % 1 ? (values[q1Pos] + values[q1Pos + 1]) / 2 : values[q1Pos]);
```

Comment:
Sounds reasonable. Looks like I missed that.
One more note I just noticed..

```
var pos = Math.floor(values.length * x); // always integer
```
so evaluation of
```
pos % 1 // always zero
```
always gives a 0 in a result. Once you fix that - check that values[pos + 1] is not out of bounds when values.length is equal to 1 (single value array).